src/Angor/Client/Services/CurrencyService.cs

```diff
@@ -0,0 +1,108 @@
+using System;
+using System.Globalization;
+using System.Net.Http;
+using System.Threading.Tasks;
+using Angor.Client.Storage;
+using System.Text.Json;
+using System.Collections.Generic;
+
+public class CurrencyService : ICurrencyService
+{
+    private readonly HttpClient _httpClient;
+    private readonly IClientStorage _storage;
+    private readonly ILogger<CurrencyService> _logger;
+
+    // Cache fields - does it work or need of redis cache/storage?
+    private Dictionary<string, (decimal rate, DateTime timestamp)> _rateCache = new();
+    private readonly TimeSpan _cacheDuration = TimeSpan.FromMinutes(10); // Cache duration of 10 minutes
+
+    public CurrencyService(HttpClient httpClient, IClientStorage storage, ILogger<CurrencyService> logger)
+    {
+        _httpClient = httpClient;
+        _storage = storage;
+        _logger = logger;
+    }
+
+    public async Task<string> GetBtcValueInPreferredCurrency(decimal btcBalance)
+    {
+        try
+        {
+            var rate = await GetBtcToPreferredCurrencyRate();
+            var currencyCode = _storage.GetCurrencyDisplaySetting();
+            var currencySymbol = GetCurrencySymbol(currencyCode);
+            var cultureInfo = new CultureInfo(GetCultureCode(currencyCode));
+            var value = btcBalance * rate;
+            return value.ToString("C2", cultureInfo).Replace(cultureInfo.NumberFormat.CurrencySymbol, currencySymbol);
+        }
+        catch (Exception ex)
+        {
+            _logger.LogError($"Error getting BTC value in preferred currency: {ex.Message}");
+            return "Error fetching value";
+        }
+    }
+
+    private async Task<decimal> GetBtcToPreferredCurrencyRate()
+    {
+        string preferredCurrency = _storage.GetCurrencyDisplaySetting().ToUpper();
+
+        // Check if we have a recent rate in the cache
+        if (_rateCache.ContainsKey(preferredCurrency) && 
+            DateTime.UtcNow - _rateCache[preferredCurrency].timestamp < _cacheDuration)
+        {
+            _logger.LogInformation($"Using cached rate for {preferredCurrency}");
+            return _rateCache[preferredCurrency].rate;
+        }
+
+        // Fetch new rate if not cached or cache is stale
+        string apiUrl = "https://mempool.space/api/v1/prices";
```

Comment:
The part that fetches should move to its own service

Reply:
I've moved it to its own service, still we need to figure out with mempool with our use is alright.
right now i made it that for each instance of angor, it will request the rates. (and will cache for 10 minutes).
still I am not sure if its requesting to much times from mempool